### ISSUE TITLE
compat.sh in 2.28 is not testing OpenSSL with TLS 1.0, TLS 1.1 or DTLS 1.0

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#6650]
In `mbedtls-2.28` as of 05dcb1300db31a1957640fce04394ca4d00db7e5:
```
$ ./tests/compat.sh -p OpenSSL -m "tls1 tls1_1 dtls1"
------------------------------------------------------------------------
PASSED (0 / 0 tests (0 skipped))
```

No TLS 1.0, TLS 1.1 and DTLS 1.0 tests run against OpenSSL, with or without mode filtering.

All versions to run when testing against mbedTLS or GnuTLS.

This doesn't apply to 3.x which doesn't have any of those protocol versions anymore.

**Comment by mpg** [comment]
Note: those tests were present in 2.28.0. (Also, the only tests that went missing between 2.28.0 and now are those.)

**Comment by mpg** [comment]
According to git bisect, the first bad commit is 01e3c37f7b88fbd46972c7204cf0e0585caefa40 which ironically has the message `compat.sh: Restore full TLS compatibility testing` - apparently this was from https://github.com/Mbed-TLS/mbedtls/pull/5660 so as I suspected that seems to be something we missed while backporting.

**Comment by ronald-cron-arm** [comment]
Yes I am on it.

**Comment by mpg** [comment]
When reviewing this sort of thing we should always run before and after and compare the outputs. I know I did that for a number of PRs reworking `compat.sh`, but apparently not this one, are at least not the backport.

Of course, it would be even better to have an automated check to tests that go missing in a PR, but that might be a bit harder to implement.

**Comment by gilles-peskine-arm** [comment]
> Of course, it would be even better to have an automated check to tests that go missing in a PR, but that might be a bit harder to implement.

My off-the-cuff idea for this is: make `compat.sh` discover what cipher suites openssl/gnutls/mbedtls support and test those, and make completeness not its job. Instrument it to record outcomes. Have outcome analysis check that every cipher suite (and combination of whatever else we think is important, e.g. USE_PSA_CRYPTO) is tested.

**Comment by ronald-cron-arm** [comment]
The fix is ready for review. Please have a look if you find some time: https://github.com/Mbed-TLS/mbedtls/pull/6654.

**Comment by mpg** [comment]
This was fixed by merging #6654
